hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -216,6 +201,47 @@ private TokenAssociation validateAndBuildAutoAssociation(
         }
     }
 
+    /**
+     * Dispatches a synthetic transaction to associate the token with the account. It will increment the usedAutoAssociations
+     * count on the account and set the token relation as automaticAssociation.
+     * This is done only if the unlimitedAutoAssociationsEnabled is enabled.
+     * @param token The token to associate with the account
+     * @param accountStore The account store
+     * @param tokenRelStore The token relation store
+     * @param context The context
+     * @param account The account to associate the token with
+     */
+    private void dispatchAutoAssociation(
+            final @NonNull Token token,
+            final @NonNull WritableAccountStore accountStore,
+            final @NonNull WritableTokenRelationStore tokenRelStore,
+            final @NonNull HandleContext context,
+            final @NonNull Account account) {
+        final var accountId = account.accountIdOrThrow();
+        final var tokenId = token.tokenIdOrThrow();
+        final var syntheticAssociation = TransactionBody.newBuilder()
+                .tokenAssociate(TokenAssociateTransactionBody.newBuilder()
+                        .account(account.accountId())
+                        .tokens(token.tokenId())
+                        .build())
+                .build();
+        // We don't need to verify signatures for this internal dispatch. So we specify the keyVerifier to null
+        context.dispatchRemovablePrecedingTransaction(
+                syntheticAssociation, SingleTransactionRecordBuilder.class, null, context.payer());
+        // increment the usedAutoAssociations count
+        final var accountModified = requireNonNull(accountStore.getAliasedAccountById(accountId))
```

Comment:
Would it make sense to introduce a `getForModify()` version? (Outside of this PR's scope.) 🤔

Reply:
Will fix this also to getForModify in next PR